React/Base/macOS/RCTUIKit.m

```diff
@@ -537,93 +537,3 @@ BOOL RCTUIViewSetClipsToBounds(RCTPlatformView *view)
 
   return clipsToBounds;
 }
-
-static BOOL RCTUIViewIsFieldEditor(RCTPlatformView *view)
-{
-  if ([view isKindOfClass:[NSText class]]) {
-    NSText *textObj = (NSText *) view;
-    return [textObj isFieldEditor];
-  }
-  return NO;
-}
-
-static BOOL RCTUIViewDescendantIsFieldEditor(RCTPlatformView *root)
-{
-  return RCTUIViewHasDescendantPassingPredicate(root, ^BOOL(RCTPlatformView *view) {
-    return RCTUIViewIsFieldEditor(view);
-  });
-}
-
-static RCTPlatformView *RCTUIViewDescendantPassingPredicate_DFS(RCTPlatformView *root, BOOL (^predicate)(RCTPlatformView *view))
-{
-  if (!root || !predicate) {
-    return nil;
-  }
-
-  if (predicate(root)) {
-    return root;
-  }
-
-  for (RCTPlatformView *subview in [root subviews]) {
-    RCTPlatformView *passingView = RCTUIViewDescendantPassingPredicate_DFS(subview, predicate);
-
-    if (passingView) {
-      return passingView;
-    }
-  }
-
-  return nil;
-}
-
-BOOL RCTUIViewHasDescendantPassingPredicate(RCTPlatformView *root, BOOL (^predicate)(RCTPlatformView *view))
-{
-  return RCTUIViewDescendantPassingPredicate_DFS(root, predicate) != nil;
-}
-
-static void RCTUIViewCalculateKeyViewLoopInternal(RCTPlatformView *root, NSMutableArray *keyViewLoop)
-{
-  for (RCTPlatformView *view in root.subviews) {
-    RCTUIViewCalculateKeyViewLoopInternal(view, keyViewLoop);
-  }
-  if ([root canBecomeKeyView]) {
-    BOOL include = YES;
-    
-    /*
-     
-     When dealing with the field editor and the key view loop, we want to take special
-     care to include the text control that is being edited, while skipping over the view
-     subtree that contains the field editor itself. The field editor manages focus/blur
-     events, and shouldn't be included in the key view loop.
-     
-     A typical construction looks like:
-     
-     NSTextField <-- Field being edited. We want to include this in the key view loop.
-       |
-       _NSKeyboardFocusClipView	<-- Field editor's superview. Not included.
-         |
-         NSTextView <-- The field editor. We do not want this in the key view loop.
-     
-     */
-    if (RCTUIViewDescendantIsFieldEditor(root)) {
-      BOOL isEditedControl = [root isKindOfClass:[NSControl class]] ? ([(NSControl *) root currentEditor] != nil) : NO;
```

Comment:
I could probably modify the special case logic I have around RCTUITextField in RCTViewManager to instead do what is described here.